src/SyncVsAsync/Services/AsyncService.cs

```diff
@@ -33,12 +33,18 @@ public class AsyncService : IAsyncService
             if (searchResultsGrid == null) return result;
 
             var searchResultsLinks = GetSearchResultsLinks(searchResultsGrid);
+
+            var PageTasks = new List<Task<HtmlDocument>>();
+
             foreach (var linkUrl in searchResultsLinks)
             {
                 result.Add(linkUrl);
                 await LoadDocumentAsync(linkUrl);
+                PageTasks.Add(LoadDocumentAsync(linkUrl));
             }
 
+            Task.WaitAll(PageTasks.ToArray());
```

Comment:
Yep, according to [Stephen Cleary's post "Async and Await"](http://blog.stephencleary.com/2012/02/async-and-await.html#guidelines), `WhenAll` should be used here.